indico/web/client/js/react/components/PublicationSwitch.jsx

```diff
@@ -0,0 +1,79 @@
+import publicationURL from 'indico-url:contributions.manage_publication';
+
+import React, {useState, useEffect} from 'react';
+import PropTypes from 'prop-types';
+import {Header, Modal, Button, Checkbox} from 'semantic-ui-react';
+import indicoAxios, {handleAxiosError} from 'indico/utils/axios';
+
+export default function PublicationSwitch({confId}) {
+  const [published, setPublished] = useState(true);
+  const [modalOpen, setModalOpen] = useState(false);
+
+  useEffect(() => {
+    const fetchPublicationSetting = async () => {
+      let response;
+      try {
+        response = await indicoAxios.get(publicationURL({confId}));
+        setPublished(response.data.published);
+      } catch (error) {
+        handleAxiosError(error);
+      }
+    };
+    fetchPublicationSetting();
+  }, [confId]);
+
+  const publish = shouldPublish => {
```

Comment:
maybe `state` or `published` would be a better name for this argument?

Reply:
Maybe `changeState` as what it really does is change the state from published to unpublished and the other way around?